nabs/callbacks.py

```diff
@@ -165,3 +165,68 @@ def stop(self, doc):
                 tags=['run_table', 'RE'],
                 title='run_table'
             )
+
+
+class BECOptionFactoryFaker():
+    """
+    Takes an instance of the BestEffortCallback, and watches for metadata
+    keys to configure the BEC for a single run
+
+    Fakes a factory function to pass into RunRouter, for the purpose
+    of generating callbacks for a RunEngine.
+
+    Factories need to have the signature: callback(name, start_doc)
+    To restore BEC state after a completed run, we must keep bec on hand
+
+    Usage:
+
+    .. code-block: python
+
+        from event_model import RunRouter
```

Comment:
What exactly is this? I must have missed this part of the bluesky ecosystem

Reply:
The RunRouter routes documents through callbacks, basically.  Here it's providing us a layer that reads the document and modify the bec before it sees documents.  You can also use it to modify the documents that callbacks see, leaving the recorded documents intact.  (I've used it in the past to help deal with file handling/saving)

Its docstring says it's an experimental feature, but it's been around for years at this point

Reply:
Okay, I think I understand what the `RunRouter` is, but not why it's used here. We aren't modifying the documents, right? Just reacting to them? What makes this formulation better than the exact same thing, but packaged as a solo thing that can be subscribed to by the RE? Does it save some verbosity + do helpful pre-callback handling?

btw I'm not trying to critique the code, I'm trying to understand for future cases

Reply:
hm, no you may be right here.  It might be simpler to just wrap the BEC in another callback that reads the documents.

Reply:
The RunRouter has paths for handling document filling, which is more involved than we need here.  It's probably more intuitive to just subclass BEC and save the state there, which I'll do now